google/cloud/bigquery/v2/minimal/internal/job_request.cc

```diff
@@ -72,29 +81,40 @@ StateFilter StateFilter::Done() {
 }
 
 std::string GetJobRequest::DebugString(TracingOptions const& options) const {
-  std::string out;
-  auto const* delim = options.single_line_mode() ? " " : "\n";
-  absl::StrAppend(&out, "GetJobRequest{", delim,
-                  "project_id=", internal::DebugString(project_id_, options),
-                  delim, "job_id=", internal::DebugString(job_id_, options),
-                  delim, internal::DebugString(location_, options), delim, "}");
-  return out;
+  auto sep = [single_line_mode = options.single_line_mode()](int indent) {
```

Comment:
I think it is a certainty that `sep()` and `quoted` will appear in many more functions, any way to refactor them?

Reply:
I like your "DebugFormatter" proposal.